lib/shared/src/sourcegraph-api/graphql/client.ts

```diff
@@ -1353,14 +1353,18 @@ export class ClientConfigSingleton {
     }
 
     public async getConfig(): Promise<CodyClientConfig | undefined> {
-        switch (this.shouldFetch()) {
-            case 'sync':
-                return this.refreshConfig()
-            // biome-ignore lint/suspicious/noFallthroughSwitchClause: This is intentional
-            case 'async':
-                this.refreshConfig()
-            case false:
-                return this.cachedClientConfig
+        try {
+            switch (this.shouldFetch()) {
+                case 'sync':
+                    return this.refreshConfig()
+                // biome-ignore lint/suspicious/noFallthroughSwitchClause: This is intentional
+                case 'async':
+                    this.refreshConfig()
+                case false:
+                    return this.cachedClientConfig
+            }
+        } catch {
+            return
```

Comment:
Would be great to log an error here for visibility
```suggestion
            logError('ClientConfigSingleton::getConfig', 'refresh failed', { verbose: error })
            return
```

Reply:
Yes, but we already log error in `this.refreshConig` (it makes log and re-through error to the caller)